### PR TITLE
Use marshaler for json logging.

### PR DESCRIPTION
**Pull request by anuraaga**
There's some regression in performance since I found the approach for writing strings to not work when actually writing to a string. We can live with it though.

After
```
Benchmark                                                              (numSpans)  Mode  Cnt        Score        Error   Units
RequestMarshalBenchmarks.marshalJson                                           16  avgt   10       54.721 ±      2.551   us/op
RequestMarshalBenchmarks.marshalJson:·gc.alloc.rate                            16  avgt   10      971.680 ±     42.850  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.alloc.rate.norm                       16  avgt   10    83613.771 ±      0.822    B/op
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Eden_Space                   16  avgt   10      977.144 ±     63.150  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Eden_Space.norm              16  avgt   10    84071.922 ±   3348.354    B/op
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Survivor_Space               16  avgt   10        0.041 ±      0.009  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Survivor_Space.norm          16  avgt   10        3.490 ±      0.721    B/op
RequestMarshalBenchmarks.marshalJson:·gc.count                                 16  avgt   10       98.000               counts
RequestMarshalBenchmarks.marshalJson:·gc.time                                  16  avgt   10       52.000                   ms
RequestMarshalBenchmarks.marshalJson                                          512  avgt   10     1647.108 ±     23.829   us/op
RequestMarshalBenchmarks.marshalJson:·gc.alloc.rate                           512  avgt   10      712.556 ±     10.067  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.alloc.rate.norm                      512  avgt   10  1847677.614 ±     33.697    B/op
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Eden_Space                  512  avgt   10      706.994 ±     62.408  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Eden_Space.norm             512  avgt   10  1833438.031 ± 165107.041    B/op
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Survivor_Space              512  avgt   10        0.065 ±      0.077  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Survivor_Space.norm         512  avgt   10      167.708 ±    197.727    B/op
RequestMarshalBenchmarks.marshalJson:·gc.count                                512  avgt   10       72.000               counts
RequestMarshalBenchmarks.marshalJson:·gc.time                                 512  avgt   10       48.000                   ms
```

Before
```
RequestMarshalBenchmarks.marshalJson                                                          16  avgt   10        49.979 ±       1.815   us/op
RequestMarshalBenchmarks.marshalJson:·gc.alloc.rate                                           16  avgt   10       991.286 ±      34.480  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.alloc.rate.norm                                      16  avgt   10     77980.835 ±       0.886    B/op
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Eden_Space                                  16  avgt   10       995.632 ±       2.215  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Eden_Space.norm                             16  avgt   10     78361.238 ±    2815.173    B/op
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Survivor_Space                              16  avgt   10         0.039 ±       0.006  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Survivor_Space.norm                         16  avgt   10         3.073 ±       0.518    B/op
RequestMarshalBenchmarks.marshalJson:·gc.count                                                16  avgt   10       100.000                counts
RequestMarshalBenchmarks.marshalJson:·gc.time                                                 16  avgt   10        60.000                    ms
RequestMarshalBenchmarks.marshalJson                                                         512  avgt   10      1864.037 ±     536.999   us/op
RequestMarshalBenchmarks.marshalJson:·gc.alloc.rate                                          512  avgt   10       584.724 ±     144.206  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.alloc.rate.norm                                     512  avgt   10   1667421.500 ±      31.495    B/op
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Eden_Space                                 512  avgt   10       570.393 ±     153.974  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Eden_Space.norm                            512  avgt   10   1627408.695 ±  181694.929    B/op
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Survivor_Space                             512  avgt   10         0.022 ±       0.017  MB/sec
RequestMarshalBenchmarks.marshalJson:·gc.churn.G1_Survivor_Space.norm                        512  avgt   10        62.799 ±      52.887    B/op
RequestMarshalBenchmarks.marshalJson:·gc.count                                               512  avgt   10        58.000                counts
RequestMarshalBenchmarks.marshalJson:·gc.time                                                512  avgt   10        40.000                    ms
```

### CODE REVIEW

File: exporters/logging-otlp/src/test/java/io/opentelemetry/exporter/logging/otlp/OtlpJsonLoggingSpanExporterTest.java [hunk 1]
```diff
@@ -118,12 +118,12 @@ void log() throws Exception {
             + "      \"traceId\": \"12340000000043211234000000004321\","
             + "      \"spanId\": \"8765000000005678\","
             + "      \"name\": \"testSpan2\","
-            + "      \"kind\": \"SPAN_KIND_CLIENT\","
+            + "      \"kind\": 3,"
```
Comment (jkwatson) [inline exporters/logging-otlp/src/test/java/io/opentelemetry/exporter/logging/otlp/OtlpJsonLoggingSpanExporterTest.java:121]: yuck. Is this really what we're supposed to be writing out with otlp json?

File: exporters/logging-otlp/src/test/java/io/opentelemetry/exporter/logging/otlp/OtlpJsonLoggingMetricExporterTest.java [hunk 2]
```diff
@@ -101,7 +101,7 @@ void log() throws Exception {
             + "          \"timeUnixNano\": \"2\","
             + "          \"asDouble\": 4.0"
             + "        }],"
-            + "        \"aggregationTemporality\": \"AGGREGATION_TEMPORALITY_CUMULATIVE\","
+            + "        \"aggregationTemporality\": 2,"
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply exporters/logging-otlp/src/test/java/io/opentelemetry/exporter/logging/otlp/OtlpJsonLoggingMetricExporterTest.java:104]: Yeah - was thinking of fixing forward it shouldn't be too tricky

[the remark this answers is not in the document]
Comment (jkwatson) [reply exporters/logging-otlp/src/test/java/io/opentelemetry/exporter/logging/otlp/OtlpJsonLoggingMetricExporterTest.java:104]: cool. do you want to create an issue for it, so we don't forget?

**Review by jkwatson — APPROVED** [review]
Good, aside from the future work to get good enum values again.

**Pull request merged**